### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Vova-Stoliar**

### CODE REVIEW

**Review by DmytroMykoliv — APPROVED** [review]
Greate work 👍

File: src/sortStudents.ts [hunk 1]
```diff
@@ -2,6 +2,73 @@
...
+export function sortStudents(
+  students: Student[],
+  sortBy: string,
+  order: string,
+): Student[] {
+  const copyStudents: Student[] = students.map((student) => {
+    return { ...student };
+  });
+
+  function reduceCall(array: number[]) : number {
```
Comment (VinnieJ-2k20) [inline src/sortStudents.ts:35]:
```suggestion
  function getAverageGrades(array: number[]) : number {
```

File: src/sortStudents.ts [hunk 2]
```diff
@@ -2,6 +2,73 @@
...
+    return array.reduce((result, currentValue) => result + currentValue);
```
Comment (VinnieJ-2k20) [inline src/sortStudents.ts:36]: Your callback calculates total grades, not average. If students will have different amount of grades, your sort will work incorrectly.

File: src/sortStudents.ts [hunk 3]
```diff
@@ -2,6 +2,73 @@
...
+          ? a[sortBy] - b[sortBy]
+          : b[sortBy] - a[sortBy];
+      }
+
+      if (SortType.Name === sortBy
+        || SortType.Surname === sortBy) {
+        return SortOrder.asc === order
+          ? a[sortBy].localeCompare(b[sortBy])
+          : b[sortBy].localeCompare(a[sortBy]);
+      }
```
Comment (VinnieJ-2k20) [inline src/sortStudents.ts:61]: 🔥

**Review by VinnieJ-2k20 — CHANGES_REQUESTED** [review]
Almost done, just need to make couple of fixes.

**Pull request closed without merging**